tools/lint/lint.py

```diff
@@ -624,6 +624,23 @@ def check_script_metadata(repo_root, path, f):
     return errors
 
 
+def check_multiple_identical_values(repo_root, path):
+    """
+    Runs lints that check for multiple identical paths
+    in all_paths_lints.
+
+    :param repo_root: the repository root
+    :param path: the path of the file within the repository
+    :returns: a list of errors found in ``path``
+    """
+
+    errors = []
+    for paths_fn in all_paths_lints:
+        if len(os.path.splitext(paths_fn)) > 0:
```

Comment:
If I assume that the input here is supposed to be a path, this is always going to be true because `splitext` is goingg to return a tuple of `path, ext`. So you need to extract the first part and keep some state to decide whether we've seen that path before. There are various ways to do that; if the list is already sorted we can just check if the current path is equal to the previous one. If we can't reply on that, the simplest approach may be to keep a dict of `{path: ext}`. If we encounter a path that's already in the dictionary we add it to a `defaultdict(list)` of `{path: [exts]}`, and then at the end we create an error for each entry in the latter dictionary, listing all the extensions that exist for the same path.